examples/experimental/dagster-airlift/examples/tutorial-example/README.md

```diff
@@ -852,3 +852,124 @@ defs = Definitions.merge(
 ```
 
 </details>
+
+## Addendum: Customizing dagster definitions
+
+`dagster-airlift` provides various pluggability points for customizing implementation behavior. If you want to change something about either
+the produced `Definitions` object or the materializations constructed from `build_defs_from_airflow_instance`, we provide additional entrypoints for this purpose.
+Whenever we want to change anything about the created `Definitions`,
+we should split out the call to `build_defs_from_airflow_instance` into these calls:
+
+1. A call to `get_resolved_airflow_defs` to resolve the initial mapped definitions.
+2. Perform your definitions-level transformations to the resolved definitions.
+3. Build the airflow polling sensor using the transformed definitions.
+   Let's say for example, that we want to modify the metadata for a spec in one of the dag assets. Here is how we could accomplish such a thing:
+
+```python
+# additional_dagster_examples/customizing_dagster.py
+from dagster import AssetSpec, Definitions
+from dagster_airlift.core import (
+    AirflowDefinitionsData,
+    AirflowInstance,
+    BasicAuthBackend,
+    build_airflow_polling_sensor_defs,
+    get_resolved_airflow_defs,
+    maps_to_dag,
+)
+
+airflow_instance = AirflowInstance(
+    auth_backend=BasicAuthBackend(
+        webserver_url="http://localhost:8080", username="admin", password="admin"
+    ),
+    name="airflow_instance_one",
+)
+
+
+def get_resolved_defs() -> Definitions:
+    """Get resolved definitions with additional metadata for a particular dag."""
+    defs = get_resolved_airflow_defs(airflow_instance=airflow_instance)
+
+    def _add_metadata(spec: AssetSpec) -> AssetSpec:
+        if maps_to_dag(spec, "my_dag"):
+            return spec._replace(metadata={"team": "my_team", **spec.metadata})
+        return spec
+
+    return defs.map_asset_specs(_add_metadata)  # type: ignore
+
+
+defs_data = AirflowDefinitionsData(
```

Comment:
Looking at this without seeing the implementation it doesn't look like `AirflowDefinitionsData` adds much value.

The "peer", "observe" mental model appears to land with folks. I suggest we leverage it in our docs and apis.

```python
peered_defs = build_peered_airflow_defs(airflow_instance)
peered_defs = peered_defs.map_asset_specs(enrich_peered_spec) 

sensors = build_airflow_polling_sensor_defs(airflow_instance, peered_defs)
```

We can use language like:

Using this model, we have a function `build_peered_airflow_defs` that creates definitions on your behalf with what we think are sensible defaults. However, before passing it `build_airflow_polling_sensor_defs`, you have the option to customize the meta information on those assets, using `map_asset_specs`.

One named and enriched as you see fit, you can pass it to `build_airflow_polling_sensor_defs`, which introspects those definitions in order to know how to translate runtime data in the Airflow instance into the appropriate events in Dagster.

`build_airflow_polling_sensor_defs` is customizable in similar fashion. The sensor creates Dagster event log events with sensible defaults, but you can supply a callback to control that exact creation and curation of those events.